File: si.21.07.gde_je_moj_ljubimac/Aplikacija/client/src/app/models/oglas.ts

export interface IOglasEnvelope {
    oglasi: IOglas[];
    oglasBroj: number;
}

export interface Slike {
    id: string;
    url: string;
    isMain: boolean;
}

export interface IOglas {

    id: string;
    naslov: string;
    dateTime: Date | null|string;
    opis: string;
    lokacija: string;
    vrsta: string;
    rasa: string;
    pol: string;
    boja: string;
    tip_Oglasa: string;
    starost: number;
    vakcinisan?: any;
    sterilisan?: any;
    papiri?: any;
    korisnikId:string;
    isKomentator: boolean;
    isHost: boolean;
    comments: IComment[];
    slike: Slike[];

}

export interface IComment {
    id: string;
    createdAt: Date;
    text: string;
    username: string;
    displayName: string;
}

export interface IOglasFormValues extends Partial<IOglas>
{}

export class OglasFormValues implements IOglasFormValues {
    id?: string = '';
    naslov: string = '';
    dateTime: Date = new Date();
    opis: string = '';
    lokacija: string = '';
    vrsta: string= '';
    rasa: string= '';
    pol: string= '';
    boja: string='';
    tip_Oglasa: string='';
    starost?: number = undefined;
    vakcinisan?: boolean = false;
    sterilisan?: boolean = false;
    papiri?: boolean = false;
    korisnikId: string = '';
    slike: { id: string; url: string; isMain: boolean }[] = [];

    constructor(init? : any) {
        Object.assign(this, init);
    }
}

File: si.21.07.gde_je_moj_ljubimac/Aplikacija/Domain/Oglas.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class Oglas
    {
        public string Id { get; set; }
        public string Naslov { get; set; }
        public DateTime DateTime { get; set; } = DateTime.UtcNow;
        public string Opis { get; set; }
        public string Lokacija { get; set; }
        public string Vrsta { get; set; }
        public string Rasa { get; set; }
        public string Pol { get; set; }
        public string Boja { get; set; }
        public string Tip_Oglasa { get; set; }
        public float? Starost { get; set; }
        public bool? Vakcinisan { get; set; }
        public bool? Sterilisan { get; set; }
        public bool? Papiri  { get; set; }
        public string KorisnikId { get; set; }
        public List<Slika> Slike { get; set; } = new List<Slika>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}